File: scala-conductr-bundle-lib/src/main/scala/com/typesafe/conductr/bundlelib/scala/Env.scala
package com.typesafe.conductr.bundlelib.scala

import com.typesafe.conductr.bundlelib.{ Env => JavaEnv }

object Env extends Env

/**
 * Standard ConductR environment vars.
 */
class Env {
  /**
   * The bundle id of the current bundle
   */
  val bundleId: Option[String] =
    Option(JavaEnv.BUNDLE_ID)

  /**
   * The URL associated with reporting status back to ConductR
   */
  val conductRStatus: Option[String] =
    Option(JavaEnv.CONDUCTR_STATUS)

  /**
   * The URL associated with locating services known to ConductR
   */
  val serviceLocator: Option[String] =
    Option(JavaEnv.SERVICE_LOCATOR)

  /**
   * The universal means of determining whether ConductR started this process
   */
  val isRunByConductR: Boolean =
    JavaEnv.isRunByConductR
}

File: java-conductr-bundle-lib/src/main/java/com/typesafe/conductr/bundlelib/java/Env.java
package com.typesafe.conductr.bundlelib.java;

import java.util.Optional;

/**
 * Standard ConductR environment vars.
 */
public class Env extends com.typesafe.conductr.bundlelib.Env {
    /**
     * The bundle id of the current bundle
     */
    public static final Optional<String> BUNDLE_ID =
            Optional.ofNullable(com.typesafe.conductr.bundlelib.Env.BUNDLE_ID);

    /**
     * The URL associated with reporting status back to ConductR
     */
    public static final Optional<String> CONDUCTR_STATUS =
            Optional.ofNullable(com.typesafe.conductr.bundlelib.Env.CONDUCTR_STATUS);

    /**
     * The URL associated with locating services known to ConductR
     */
    public static final Optional<String> SERVICE_LOCATOR =
            Optional.ofNullable(com.typesafe.conductr.bundlelib.Env.SERVICE_LOCATOR);
}
